pruning/architecture/pruning_methods/iterators.py

```diff
@@ -0,0 +1,90 @@
+from typing import Generator
+
+from config.iterators import PruningIterator
+import numpy as np
+
+
+def iterative(start: float, end: float, step: float) -> Generator[float, None, None]:
+    """
+    Generates a sequence of values from start to end with a given step size.
+
+    Args:
+        start (float): The starting value of the sequence.
+        end (float): The ending value of the sequence.
+        step (float): The step size between consecutive values.
+
+    Yields:
+        float: The next value in the sequence.
+
+    """
+    num_steps = int((end - start) / step) - 1
+
+    if start != 0:
+        yield start
+
+    for _ in np.linspace(start, end, num_steps):
+        yield step
+
+
+def one_shot(start: float, end: float, step: float) -> Generator[float, None, None]:
+    """Generates a single value and yields it.
+
+    Args:
+        start (float): The starting value.
+        end (float): The ending value.
+        step (float): The step size.
+
+    Yields:
+        float: The end value.
+    """
+    yield end
+
+
+def logarithmic(start: float, end: float, step: float) -> Generator[float, None, None]:
+    """Generates logarithmically spaced values between a start and end value.
+
+    Args:
+        start (float): The starting value.
+        end (float): The ending value.
+        step (float): The step size.
+
+    Yields:
+        float: The next logarithmically spaced value.
+    """
+    num_values = int((end - start) / step)
+    total_sum = end - start
+
+    values = np.logspace(np.log10(start), np.log10(num_values), num=num_values, base=10)
+
+    values *= total_sum / np.sum(values)
+
+    yield start
+    for value in reversed(values):
+        yield round(value, 3)
+
+
+def get_iterator(iterator: PruningIterator) -> Generator[float, None, None]:
```

Comment:
I guess construct_iterator will be more consistent naming with the rest of the functions like construct_optimizer etc..

Reply:
done